dkg/frost/frost.go

```diff
@@ -0,0 +1,113 @@
+package frost
+
+import (
+	"encoding/json"
+
+	"github.com/bloxapp/ssv-spec/dkg"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"github.com/coinbase/kryptology/pkg/dkg/frost"
+	"github.com/herumi/bls-eth-go-binary/bls"
+)
+
+var thisCurve = curves.BLS12381G1()
+
+type FROST struct {
+	identifier dkg.RequestID
+	network    dkg.Network
+	operatorID uint32
+	operators  []uint32
+	threshold  uint32
+	party      *frost.DkgParticipant
+
+	validatorPK    []byte
+	operatorShares map[uint32]*bls.SecretKey
+
+	keygenMessage map[uint32]KeygenMessage
+}
+
+type KeygenMessage struct {
+	Round1 Round1Message
+	Round2 Round2Message
+}
+
+func New(
+	requestID dkg.RequestID,
+	network dkg.Network,
+	i, t uint32,
+	operators []uint32,
+) dkg.KeyGenProtocol {
+
+	otherOperators := make([]uint32, 0)
+	for _, operatorID := range operators {
+		if operatorID == i {
+			continue
+		}
+		otherOperators = append(otherOperators, operatorID)
+	}
+
+	ctx := "string to prevent replay attacks"
+	party, _ := frost.NewDkgParticipant(i, uint32(len(operators)), ctx, thisCurve, otherOperators...)
+
+	return &FROST{
+		identifier: requestID,
+		network:    network,
+		operatorID: i,
+		threshold:  t,
+		operators:  operators,
+		party:      party,
+	}
+}
+
+func (frost *FROST) Start(init *dkg.Init) error {
+	// TODO: use init message to initialize number of operators and threshold
+	bCastMessage, p2pMessages, err := frost.party.Round1(nil)
```

Comment:
In our case, there will be one more round before `Round1` which will send `PreparationMessage`.

Reply:
We can call `frost.NewDkgParticipant` here to use the info carried in `init` Message.